### PR TITLE
Feature:team、department、user function union

### PR DESCRIPTION
**Pull request by MayueCif**

### CODE REVIEW

File: src/Contracts/Masa.Auth.Contracts.Admin/Infrastructure/Enums/GenderTypes.cs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Masa.Auth.Contracts.Admin.Infrastructure.Enums;
+
+public enum GenderTypes
+{
+    Male =1,
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Infrastructure/Enums/GenderTypes.cs:5]: code format

File: src/Contracts/Masa.Auth.Contracts.Admin/Masa.Auth.Contracts.Admin.csproj [hunk 2]
```diff
@@ -7,9 +7,10 @@
   </PropertyGroup>
 
   <ItemGroup>
+    <PackageReference Include="EnumDesc" Version="0.2.0" />
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Masa.Auth.Contracts.Admin.csproj:10]: We have this.

File: src/Contracts/Masa.Auth.Contracts.Admin/Subjects/AddUserDto.cs [hunk 3]
```diff
@@ -28,6 +28,8 @@ public class AddUserDto
 
     public string Password { get; set; }
 
+    public GenderTypes GenderType { get; set; }
```
Comment (doddgu) [inline src/Contracts/Masa.Auth.Contracts.Admin/Subjects/AddUserDto.cs:31]: GenderType -> Gender

File: src/Services/Masa.Auth.Service.Admin/Application/Permissions/QueryHandler.cs [hunk 4]
```diff
@@ -52,6 +52,19 @@ private async Task GetRoleSelectAsync(RoleSelectQuery query)
         query.Result = await _authDbContext.Set<Role>().Select(r => new RoleSelectDto(r.Id, r.Name)).ToListAsync();
     }
 
+    #region Permission
+
+    [EventHandler]
+    public async Task PermissionTypesQueryAsync(PermissionTypesQuery permissionTypesQuery)
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Application/Permissions/QueryHandler.cs:58]: Why have to use async?

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Users/UpdateUserDialog.razor [hunk 5]
```diff
@@ -1,119 +1,135 @@
...
-                    </MAvatar>
-                    <MTextField @bind-Value="User.DisplayName" Class="mt-0" HideDetails="@("auto")" Placeholder="@T("Give him a nickname!")" />
+<SheetDialog Value="Visible" ValueChanged=UpdateVisible Title="@T("Update User")">
+    <MForm Model=User EnableValidation @ref="Form">
+        <div style="padding-left:204px;padding-right:204px;padding-bottom:24px;padding-top:24px;height:100%;width: 100%;position: absolute;">
+             <div style="position: relative;" class="full-height">    
+                <div class="d-flex">
+                    <div class="mx-auto text-center" style="width:212px;">
+                        <MAvatar Size=80>
+                            <img src="/_content/Masa.Auth.Web.Admin.Rcl/img/subject/user.svg" />
```
Comment (doddgu) [inline src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Users/UpdateUserDialog.razor:10]: cdn

File: src/Services/Masa.Auth.Service.Admin/Masa.Auth.Service.Admin.csproj [hunk 6]
```diff
@@ -11,25 +11,26 @@
 	</PropertyGroup>
 	
 	<ItemGroup>
-		<PackageReference Include="Masa.BuildingBlocks.Dispatcher.Events" Version="0.3.2" />
-		<PackageReference Include="Masa.Contrib.Configuration" Version="0.3.2" />
-		<PackageReference Include="Masa.Contrib.Data.Contracts.EF" Version="0.3.2" />
-		<PackageReference Include="Masa.Contrib.Dispatcher.Events" Version="0.3.2" />
-		<PackageReference Include="Masa.Contrib.Dispatcher.IntegrationEvents.EventLogs.EF" Version="0.3.2" />
+		<PackageReference Include="Masa.BuildingBlocks.Dispatcher.Events" Version="0.4.0-prevew.1" />
```
Comment (doddgu) [inline src/Services/Masa.Auth.Service.Admin/Masa.Auth.Service.Admin.csproj:14]: preview2

File: src/Contracts/Masa.Auth.Contracts.Admin/Masa.Auth.Contracts.Admin.csproj (hunk 2 above)
Comment (MayueCif) [reply src/Contracts/Masa.Auth.Contracts.Admin/Masa.Auth.Contracts.Admin.csproj:10]: support by @capdiem

File: src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Users/UpdateUserDialog.razor (hunk 5 above)
Comment (15168440402) [reply src/Web/Masa.Auth.Web.Admin.Rcl/Pages/Subjects/Users/UpdateUserDialog.razor:10]: This is written temporarily, the latter is not a dead url

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**